### PR TITLE
Refactor mask

### PR DESCRIPTION
**Pull request by chewxy**
refactored the masked tensor construction - removed the need for (...[]bool), instead, added an additional `addMask()` method.

### CODE REVIEW

File: tensor/dense_io.go [hunk 1]
```diff
@@ -471,92 +471,93 @@ func (t *Dense) GobDecode(p []byte) (err error) {
...
+		t.fromSlice(data)
 	case reflect.String:
 		var data []string
 		if err = decoder.Decode(&data); err != nil {
 			return
 		}
-		t.fromSlice(data, mask)
+		t.fromSlice(data)
 	}
+	t.addMask(mask)
```
Comment (chewxy) [inline tensor/dense_io.go]: This is where all the addMasks are replaced @kabaka0

**Comment by chewxy** [comment]
I'd love your thoughts on this @kabaka0 . So far I have ran all the tests and it all passed. Is there a particular reason why you added a `(...[]bool)` option for `fromSlice`?

File: tensor/dense_io.go (hunk 1 above)
Comment (kabaka0) [inline tensor/dense_io.go]: This looks fine.

**Comment by kabaka0** [comment]
That particular construction was probably a legacy of some code I had previously written before submitting the pull request. I agree that having a separate addMask() method probably makes it less confusing for users that never have reason to use the mask.

**Comment by chewxy** [comment]
Cheers mate.

**Pull request merged**